metadata/src/main/java/org/apache/kafka/controller/ClientQuotaControlManager.java

```diff
@@ -170,8 +170,10 @@ private void alterClientQuotaEntity(
             }
         });
 
-        outputRecords.addAll(newRecords);
-        outputResults.put(entity, ApiError.NONE);
+        // Only add the records to outputRecords if there were no errors
```

Comment:
This is a good fix.

However, I think it would be better just to return immediately after setting the error, rather than waiting until the end of the function.  That's consistent with how we handle errors at the top of this function, and in other manager classes.  It makes it clear that only one error can be set for each entity.